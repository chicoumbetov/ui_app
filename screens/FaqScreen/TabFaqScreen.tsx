/**
 * FAQ page with mock data
 *
 * @author: Shynggys UMBETOV
 */

import * as React from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';

import { Layout } from '@ui-kitten/components';

import { createStackNavigator } from '@react-navigation/stack';
import { DrawerActions } from '@react-navigation/native';
import { AntDesign } from '@expo/vector-icons';
import Faq from './Components/Faq';
import HeaderLeftOpenDrawerNavigation from '../../navigation/HeaderLeftOpenDrawerNavigation';

const Stack = createStackNavigator();

const TabFaqScreen = ({ navigation }) => (
  <Layout style={styles.container}>
    <Stack.Navigator>
      <Stack.Screen
        name="Faq"
        component={Faq}
        options={{
          headerTitle: false,
          headerShown: true,
          headerStyle: {
            height: 120,
          },
          headerRightContainerStyle: {
            marginRight: 18,
          },
          headerLeftContainerStyle: {
            marginLeft: 13,
          },
          headerLeft: () => (
            <TouchableOpacity onPress={() => {
              navigation.dispatch(DrawerActions.toggleDrawer());
            }}
            >
              <AntDesign name="arrowleft" size={30} style={{ color: '#b5b5b5', marginLeft: 20 }} />
            </TouchableOpacity>
          ),
          headerRight: () => (
            <HeaderLeftOpenDrawerNavigation navigation={navigation} />
          ),
        }}
      />
    </Stack.Navigator>
  </Layout>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'rgba(246, 246, 246, 0.5)',
  },
});

export default TabFaqScreen;
