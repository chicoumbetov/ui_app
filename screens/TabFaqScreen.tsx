import * as React from 'react';
import { StyleSheet } from 'react-native';

import { Layout } from '@ui-kitten/components';
import Faq from '../components/Faq/Faq';
import faqQandA from '../mockData/faqQandA';

const TabFaqScreen = () => (
  <Layout style={styles.container}>
    <Faq faqQandAs={faqQandA} />
  </Layout>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'rgba(246, 246, 246, 0.5)',
  },
});

export default TabFaqScreen;
