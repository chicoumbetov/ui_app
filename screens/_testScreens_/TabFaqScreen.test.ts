import renderer from 'react-test-renderer';

import TabFaqScreen from '../FaqScreen/TabFaqScreen';

describe('<TabFaqScreen />', () => {
  it('renders correctly across screens', () => {
    const tree = renderer
      .create(
        '<TabFaqScreen />',
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

it('<TabFaqScreen />', () => {
  expect(TabFaqScreen);
});
