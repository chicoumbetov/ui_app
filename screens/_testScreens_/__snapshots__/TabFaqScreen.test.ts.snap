// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabFaqScreen /> renders correctly across screens 1`] = `"<Faq />"`;
